src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -30,11 +30,14 @@
 
   <PropertyGroup>
     <FeaturePortableThreadPool Condition="'$(FeaturePortableThreadPool)' == ''">false</FeaturePortableThreadPool>
-    <FeaturePortableThreadPool Condition="'$(TargetsUnix)' == 'true' or '$(TargetsBrowser)'=='true'">true</FeaturePortableThreadPool>
+    <FeaturePortableThreadPool Condition="'$(TargetsUnix)' == 'true' or '$(TargetsBrowser)'=='true' or '$(TargetsWasi)'=='true'">true</FeaturePortableThreadPool>
   </PropertyGroup>
   <PropertyGroup>
     <FeaturePortableTimer Condition="'$(FeaturePortableTimer)' == ''">false</FeaturePortableTimer>
-    <FeaturePortableTimer Condition="'$(TargetsUnix)' == 'true' or '$(TargetsBrowser)'=='true'">true</FeaturePortableTimer>
+    <FeaturePortableTimer Condition="'$(TargetsUnix)' == 'true' or '$(TargetsBrowser)'=='true' or '$(TargetsWasi)'=='true'">true</FeaturePortableTimer>
+  </PropertyGroup>
+  <PropertyGroup>
+    <FeatureWasmThreads Condition="'$(TargetsWasi)' == 'true'">true</FeatureWasmThreads>
```

Comment:
Am I right to understand threads are basically only to have various Unix ptthread APIs available for compilation?

If so, why does managed code need to enable them?

Reply:
We get this error:
```
E:\GitHub\runtimelab\src\libraries\System.Private.CoreLib\src\System\Threading\ThreadPoolBoundHandleOverlapped.cs(15,27): error CS0246: The type or namespace
 name 'PreAllocatedOverlapped' could not be found (are you missing a using directive or an assembly reference?) [E:\GitHub\runtimelab\src\coreclr\nativeaot\S
ystem.Private.CoreLib\src\System.Private.CoreLib.csproj]
E:\GitHub\runtimelab\src\libraries\System.Private.CoreLib\src\System\Threading\ThreadPoolBoundHandle.cs(216,66): error CS0246: The type or namespace name 'Pr
eAllocatedOverlapped' could not be found (are you missing a using directive or an assembly reference?) [E:\GitHub\runtimelab\src\coreclr\nativeaot\System.Pri
vate.CoreLib\src\System.Private.CoreLib.csproj]
E:\GitHub\runtimelab\src\libraries\System.Private.CoreLib\src\System\Threading\ThreadPoolBoundHandleOverlapped.cs(21,111): error CS0246: The type or namespac
e name 'PreAllocatedOverlapped' could not be found (are you missing a using directive or an assembly reference?) [E:\GitHub\runtimelab\src\coreclr\nativeaot\
System.Private.CoreLib\src\System.Private.CoreLib.csproj]
```
If we try to turn off `FeaturePortableThreadPool` we get errors like
```
E:\GitHub\runtimelab\src\libraries\System.Private.CoreLib\src\System\Threading\ThreadPoolWorkQueue.cs(1381,23): error CS0246: The type or namespace name 'Reg
isteredWaitHandle' could not be found (are you missing a using directive or an assembly reference?) [E:\GitHub\runtimelab\src\coreclr\nativeaot\System.Privat
e.CoreLib\src\System.Private.CoreLib.csproj]
```
I can work through this in more detail if you like?

Reply:
It would be great to get to the bottom of this. The overriding concern is that we shouldn't enable features that don't work.

Reply:
Threads are clearly prevalent across `System.Private.CoreLib` and not behind flags https://github.com/dotnet/runtimelab/blob/c22ff170b51086a500c378ef31eaece19a0987f7/src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems#L1129-L1213 however if we turn off the features, we can compensate with some throwing stubs in nativeaot.  There's quite a few to get past the ApiCompat check and the fact that some other parts of the code reference threads, `SafeFileHandle` being one example.